classes/01class/exercises/c01-aws01/albertoeks/README.md

```diff
@@ -0,0 +1,340 @@
+                    "VpcId": "vpc-c6aba3a1",
+                    "InterfaceType": "interface"
+                }
+            ],
+            "RootDeviceName": "/dev/sda1",
+            "RootDeviceType": "ebs",
+            "SecurityGroups": [
+                {
+                    "GroupName": "SG-PUBLIC",
+                    "GroupId": "sg-0db7b9ef1a468cd10"
+                }
+            ],
+            "SourceDestCheck": true,
+            "StateReason": {
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "059642851259",
+    "ReservationId": "r-0119ff3890b14d90a"
+}
+
+# Change the permission for AlbertoKeyPair.pem
+$ chmod 400 AlbertoKeyPair.pem
+
+#Access the EC2 instance via ssh (checked the public IP on AWS console)
+$ ssh -i AlbertoKeyPair.pem ubuntu@13.54.171.156
+
+Welcome to Ubuntu 18.04.4 LTS (GNU/Linux 4.15.0-1065-aws x86_64)
+
+ * Documentation:  https://help.ubuntu.com
+ * Management:     https://landscape.canonical.com
+ * Support:        https://ubuntu.com/advantage
+
+  System information as of Sat Jul 18 06:27:45 UTC 2020
+
+  System load:  0.02              Processes:           88
+  Usage of /:   13.9% of 7.69GB   Users logged in:     0
+  Memory usage: 18%               IP address for eth0: 172.31.19.236
+  Swap usage:   0%
+
+ * Canonical Livepatch is available for installation.
+   - Reduce system reboots and improve kernel security. Activate at:
+     https://ubuntu.com/livepatch
+
+0 packages can be updated.
+0 updates are security updates.
+
+
+
+The programs included with the Ubuntu system are free software;
+the exact distribution terms for each program are described in the
+individual files in /usr/share/doc/*/copyright.
+
+Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
+applicable law.
+
+To run a command as administrator (user "root"), use "sudo <command>".
+See "man sudo_root" for details.
+
+ubuntu@ip-172-31-19-236:~$
+```
+
+- Create another EC2, also using the CLI instance in the same VPC but with private address only and connect to it.
+
+```bash
+#Create security group for second EC2
+$ aws ec2 create-security-group --group-name SG-PRIVATE --description "SG-PRIVATE for second EC2"
+{
+    "GroupId": "sg-04a8651999ca43673"
+}
+
+#Enable SSH for security group
+$ aws ec2 authorize-security-group-ingress --group-id sg-04a8651999ca43673 --protocol tcp --port 22 --cidr 0.0.0.0/0 --region ap-southeast-2
```

Comment:
Instead of allowing any ip address to connect (0.0.0.0/0). You could only allow connection from other security group replacing `--cidr` to `--source-group <sg-id>`.

```bash
aws ec2 authorize-security-group-ingress \
    --group-id sg-04a8651999ca43673 \
    --protocol tcp \
    --port 22 \
    --source-group sg-0db7b9ef1a468cd10 # this is your public SG ID
```

Reply:
Nice, thanks for the advice!